G-Earth/src/main/java/gearth/services/extension_handler/extensions/implementations/network/NetworkExtensionsProducer.java

```diff
@@ -2,85 +2,136 @@
+     */
+    private static final int PACKET_HEADER_SIZE = Integer.BYTES + Short.BYTES;
+
+    /**
+     * Represents the maximum number of bytes in the body of an incoming packet.
+     * <p>
+     * Used as a form of validation for packets, prevents other Apps that connect
+     * with the server from sending unexpected data and inexplicably causing huge byte array allocations.
+     * <p>
+     * Since the server only accepts {@link NetworkExtensionInfo.INCOMING_MESSAGES_IDS#EXTENSIONINFO} packets,
+     * this value is calculated based on that packet.
+     */
+    private static final int MAX_PACKET_BODY_SIZE = (MAX_STRING_SIZE * 6) + (BOOLEAN_SIZE * 4);
+
+    /**
+     * The port at which the {@link #serverSocket} is listening for incoming connections.
+     */
     public static int extensionPort = -1;
 
-
     private ServerSocket serverSocket;
 
     @Override
     public void startProducing(ExtensionProducerObserver observer) {
-//        serverSocket = new ServerSocket(0);
-        int port = 9092;
-        boolean serverSetup = false;
-        while (!serverSetup) {
-            serverSetup = createServer(port);
-            port++;
-        }
 
+        /*
+        Initialise the serverSocket at the argued port.
+         */
+        int port = PORT_ONSET;
+        while (!createServer(port))
+            ++port;
 
+        /*
+        Start connection listener thread.
+         */
         new Thread(() -> {
+
             try {
+
                 while (!serverSocket.isClosed()) {
-                    Socket extensionSocket = serverSocket.accept();
+
+                    // accept a new connection
+                    final Socket extensionSocket = serverSocket.accept();
                     extensionSocket.setTcpNoDelay(true);
 
+                    /*
+                    Start client session handler thread.
+                     */
                     new Thread(() -> {
+
                         try {
+
+                            // write INFOREQUEST packet to client
                             synchronized (extensionSocket) {
                                 extensionSocket.getOutputStream().write((new HPacket(NetworkExtensionInfo.OUTGOING_MESSAGES_IDS.INFOREQUEST)).toBytes());
                             }
 
-                            InputStream inputStream = extensionSocket.getInputStream();
-                            DataInputStream dIn = new DataInputStream(inputStream);
+                            final DataInputStream dIn = new DataInputStream(extensionSocket.getInputStream());
 
+                            // listen to incoming data from client
                             while (!extensionSocket.isClosed()) {
 
-                                int length = dIn.readInt();
-                                byte[] headerandbody = new byte[length + 4];
+                                if (dIn.available() < PACKET_HEADER_SIZE)
+                                    continue;
```

Comment:
since we're trying to fix an issue with random programs connecting with the g-earth extension port, incase the program doesn't send bytes after the tcp connection is established (perhaps it waits for the server to respond or whatever), g-earth will be stuck in an infinite loop here 

I think we can skip this check, and immediately execute 
```
final int bodyLength = dIn.readInt() - Short.BYTES;
final short headerId = dIn.readShort();
```

Iirc those methods are blocking until the bytes are available